src/main/groovy/de/dkfz/roddy/execution/io/AsyncExecutionResult.groovy

```diff
@@ -0,0 +1,66 @@
+package de.dkfz.roddy.execution.io
+
+import java.util.concurrent.Future
+import java.util.concurrent.CompletableFuture
+
+/** Like ExecutionResult, but taking Futures of the exit code (usually the actual process) and the standard output and
+ *  error streams. Calling any of the methods will block until the process is finished!
+ */
+class AsyncExecutionResult extends ExecutionResult {
+
+    private final Future<Integer> exitCodeF
+    private final Future<List<String>> stdoutF
+    private final Future<List<String>> stderrF
+    private final Future<Boolean> successfulF
+
+    /** The AsyncExecutionResult takes Futures of a number of values. Calling any of the methods of the superclass
+     *  that return the values will block until the result is available.
+     *
+     * @param process
+     * @param stdout
+     * @param stderr
+     * @param successful   Can be a future. By default (when set to null), it will be set success if exitCode == 0.
+     */
+    AsyncExecutionResult(CompletableFuture<Integer> process,
+                         CompletableFuture<List<String>> stdout,
+                         CompletableFuture<List<String>> stderr,
+                         CompletableFuture<Boolean> successful = null) {
+        super(true, 0, [], "") // These are only stub-values.
```

Comment:
You are completely right, but this actually is a big problem. The class is used in Roddy core,BatchEuphoria, COWorkflows 1.2.3 + 1.2.20 and the alignment workflow. A bytecode-level change of the class is actually API-breaking and would require a re-release of these packages, which is particularly problematic for the COWorkflows plugin and the alignment plugin because of all the versions. I think this is not worth the effort. I will  add a comment here and make an issue for the next major release of RoddyToolLib -- which will probably never happen though.